langgraph/src/channels/base.ts

```diff
@@ -85,9 +102,9 @@ export async function createCheckpoint<ValueType>(
   const newCheckpoint: Checkpoint = {
     v: 1,
     ts: new Date().toISOString(),
-    channelValues: { ...checkpoint.channelValues },
-    channelVersions: { ...checkpoint.channelVersions },
-    versionsSeen: { ...checkpoint.versionsSeen },
+    channelValues: deepCopy(checkpoint.channelValues),
```

Comment:
`checkpoint.channelValues` and `checkpoint.channelVersions` don't need to be deep copied. They can be set to an empty object so previous channel values aren't carried over. This is equivalent to Python implementation.